### PR TITLE
fix execute request/response serialization/deserialization

### PR DESCRIPTION
**Pull request by ylwu-amzn**
Signed-off-by: Yaliang Wu <ylwu@amazon.com>

### Description
Fix execute request/response serialization/deserialization;
Update how to add new function doc.
 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/ml/common/parameter/AnomalyLocalizationOutput.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package org.opensearch.ml.engine.algorithms.anomalylocalization;
+package org.opensearch.ml.common.parameter;
```
Comment (Zhangxunmt) [inline common/src/main/java/org/opensearch/ml/common/parameter/AnomalyLocalizationOutput.java:6]: I have been wondering this ml.common.parameter directory is bloating quickly and it already includes so many files. Should we considering using subdirectories like model, task, <algorithmA>, <algorithmB>, etc?

Comment (ylwu-amzn) [reply common/src/main/java/org/opensearch/ml/common/parameter/AnomalyLocalizationOutput.java:6]: Yes, that's the plan. The refactoring may impact quite a lot places. Let's finish the testing first, then move these classes to different packages.

File: common/src/main/java/org/opensearch/ml/common/transport/execute/MLExecuteTaskRequest.java [hunk 2]
```diff
@@ -31,21 +33,25 @@
 @ToString
 public class MLExecuteTaskRequest extends ActionRequest {
 
+    FunctionName functionName;
```
Comment (Zhangxunmt) [inline common/src/main/java/org/opensearch/ml/common/transport/execute/MLExecuteTaskRequest.java:36]: Is this FunctionName field the key part that triggered the bug? I see that this triggered most of the required changes in the PR.

**Review by Zhangxunmt — APPROVED** [review]

File: common/src/main/java/org/opensearch/ml/common/transport/execute/MLExecuteTaskRequest.java (hunk 2 above)
Comment (ylwu-amzn) [reply common/src/main/java/org/opensearch/ml/common/transport/execute/MLExecuteTaskRequest.java:36]: Yes, we need `functionName` to tell which class to initialize for this algorithm.

**Pull request merged**